### DISCUSSION TITLE
How to persist `tagName` of a custom component

### DISCUSSION THREAD

**Discussion by padcom** [GrapesJS/grapesjs#5411] [Q&A]
Hi, I'm trying to create a set of custom components that will render into a wizard. I wanted to use `ul` and `li` for some of the parts, but the `tagName` is missing from the project data:

```javascript
ed.DomComponents.addType('wizard-tab', {
  model: {
    defaults: {
      tagName: 'li',
      ....
    },
  },
})
```

How can I make a component force to store the tag name in the resulting project data? I use the project data later on to render some outputs on the backend and I am not currently using the `toHTML()` function to render it from GrapesJS. Any advice, please?

**Comment by artf** [comment]
Properties like `tagName` (and other string/number/boolean types) are skipped from the JSON output if the value is the same in the component's definition.
You should be able to extend default component types with your own implementation of [toJSON](https://github.com/GrapesJS/grapesjs/blob/7ac146ce66695570226567366d70418224e0280b/src/dom_components/model/Component.ts#L1627)
